packages/rules/src/consistentVersions.ts

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright 2020 Palantir Technologies, Inc.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ *
+ */
+
+import { Context, RuleModule } from "@monorepolint/core";
+import { mutateJson, PackageJson } from "@monorepolint/utils";
+import * as r from "runtypes";
+import { coerce } from "semver";
+
+export const Options = r.Record({
+  matchDependencyVersions: r.Dictionary(r.String),
+});
+
+type Options = r.Static<typeof Options>;
+
+export const consistentVersions: RuleModule<typeof Options> = {
+  check: checkConsistentVersions,
+  optionsRuntype: Options,
+};
+
+function checkConsistentVersions(context: Context, options: Options) {
+  for (const [dependencyPackageName, expectedPackageDependencyValue] of Object.entries(
+    options.matchDependencyVersions
+  )) {
+    ensurePackageIsCorrectVersion(context, dependencyPackageName, expectedPackageDependencyValue);
+  }
+}
+
+const ensurePackageIsCorrectVersion = (
+  context: Context,
+  dependencyPackageName: string,
+  expectedPackageDependencyValue: string
+) => {
+  const packageJson = context.getPackageJson();
+  const packageJsonPath = context.getPackageJsonPath();
+
+  const expectedPackageDependencyVersion = coerce(expectedPackageDependencyValue);
+  if (expectedPackageDependencyVersion == null) {
+    throw new Error(
+      `Malformed expected package dependency version defined in monorepolint configuration: ${dependencyPackageName} @ '${expectedPackageDependencyValue}'`
+    );
+  }
+
+  const actualPackageDependencyValue = packageJson.dependencies && packageJson.dependencies[dependencyPackageName];
```

Comment:
prettier was not happy about `packageJson.dependencies?.[dependencyPackageName]`

Reply:
huh, why not?

Reply:
<img width="655" alt="Screen Shot 2020-06-05 at 2 45 39 PM" src="https://user-images.githubusercontent.com/8195549/83912136-589fd400-a73b-11ea-8ae4-c6bb43ead481.png">
<img width="733" alt="Screen Shot 2020-06-05 at 2 46 01 PM" src="https://user-images.githubusercontent.com/8195549/83912149-5a699780-a73b-11ea-937d-764fb107de07.png">

Reply:
this is a typescript issue, this repo's version of typescript is too old:
https://github.com/monorepolint/monorepolint/blob/2d6068c00482889f16169cc1885d331fc5d66793/package.json#L61